deliver/spec/upload_screenshots_spec.rb

```diff
@@ -0,0 +1,73 @@
+require 'spec_helper'
+require 'fakefs/spec_helpers'
+require 'stubs/upload_stubbing'
+
+describe Deliver::UploadScreenshots do
+  include FakeFS::SpecHelpers
+
+  before :each do
+    @root = '/tmp/screenshots/'
+    FileUtils.mkdir_p(@root)
+
+    (1..5).each do |i|
+      file_path = File.join(@root, "scr_#{i}.jpg")
+      File.open(file_path, 'w') { |file| file.write("#{i}") }
+    end
+  end
+
+  def mock_local_upload(sequence)
+    @order = 0
+    to_upload = sequence.map do |i|
+      file_path = File.join(@root, "scr_#{i}.jpg")
+      md5 = Spaceship::Utilities.md5digest(file_path)
+      file = Screenshot.new({file_path: file_path, language: 'en-US', original_file_name: "ftl_#{md5}_scr_#{i}.jpg", sort_order: @order})
+      @order += 1
+      file
+    end
+    to_upload
+  end
+
+  let(:options) { { app: FakeApp.new } }
+
+  context "Deleting screenshots from ITC" do
+    it "should delete screenshot from ITC when order had been changed" do
+      # 1st and 2nd changed places. Expecting deletion of both
+      expect do
+        Deliver::UploadScreenshots.new.upload(options, mock_local_upload([2, 1, 3, 4]))
```

Comment:
do we need to create a new UploadScreenshots instance every time ? Let's test with the same, to make sure it doesn't keep state that it shouldn't. (I am not so sure of the way the @local_checksums are kept. Maybe they should be passed around instead. Less state is sometimes better.